tools/Google.Cloud.Tools.SbomGenerator/Program.cs

```diff
@@ -130,11 +115,26 @@ private static Spdx SpdxFrom(string nupkgPath)
     /// </summary>
     /// <param name="nuspecXmlStream">xml content</param>
     /// <returns>metadata</returns>
-    private static NuspecPackage GetMetadata(Stream nuspecXmlStream)
+    private static NuspecPackage GetNuspecMetadata(string nupkgPath)
     {
         XmlSerializer xmlSerializer = new XmlSerializer(typeof(NuspecPackage));
+        using var zipFile = ZipFile.OpenRead(nupkgPath);
+        using var nuspecStream = zipFile.Entries.Single(entry => Path.GetExtension(entry.Name).ToLower() == ".nuspec").Open();
+        return xmlSerializer.Deserialize(nuspecStream) as NuspecPackage;
+    }
+
+    private static async Task<(string hash, string uuid)> GetSha256AndGuidAsync(string nupkgPath)
+    {
+        // Compute the SHA-256 hash of the .nupkg file.
+        using var sha256 = SHA256.Create();
+        using var file = File.OpenRead(nupkgPath);
+
+        var hashBytes = await sha256.ComputeHashAsync(file);
+
+        // Use the SHA-256 hash to create a deterministic uuid.
+        var uuid = Uuid5FromHash(hashBytes);
 
-        return xmlSerializer.Deserialize(nuspecXmlStream) as NuspecPackage;
+        return (BitConverter.ToString(hashBytes).Replace("-", ""), uuid.ToString());
```

Comment:
Remove `()` surrounding the expression? (Can do that later of course.)

Reply:
It's a tuple that I'm returnig here, the hash text, and the uuid built from the hash bytes.